src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java

```diff
@@ -531,6 +533,11 @@ public boolean setPlayerRank(final UUID id, final Rank rank, final World world)
     private boolean addPlayer(@NotNull final GameProfile gameprofile, final Rank rank)
     {
         @NotNull final Player p = new Player(gameprofile.getId(), gameprofile.getName(), rank);
+
+        if(players.containsKey(p.id))
```

Comment:
I don't know why but for some reason the rank never got updated.
if you do "addOfficer" and you're friend you're not upgraded to officer.